config/dynamic.c

```diff
@@ -534,6 +540,9 @@ run_dynamic_config(const char *server_path, struct dynamic_config *dy_conf)
 	if (ret < 0)
 		goto free_sock;
 
+	gk_conf_hold(gk_conf);
+	dy_conf->gk = gk_conf;
```

Comment:
Same kind of issue here -- we only want to execute this statements when `gatekeeper` is being run as a Gatekeeper server.